### PR TITLE
feat: es6 support with types

### PR DESCRIPTION
**Pull request by dinesh-kumar-allen**

### CODE REVIEW

File: packages/material-tailwind-react/package.json [hunk 1]
```diff
@@ -1,37 +1,37 @@
 {
-  "name": "@material-tailwind/react",
+  "name": "@allen-career-institute/material-tailwind-react",
   "homepage": "https://material-tailwind.com",
-  "version": "2.1.10",
+  "version": "1.0.0",
```
Comment (aakash-allen-forced) [inline packages/material-tailwind-react/package.json]: would suggest to keep version number as 2.2.x, put a note in readme that we forked from 2.1.10 version of original library.

Comment (dinesh-kumar-allen) [reply packages/material-tailwind-react/package.json]: Yes updated as per recommendation

File: packages/material-tailwind-react/src/components/Alert/index.tsx [hunk 2]
```diff
@@ -55,8 +55,7 @@ export interface AlertProps extends Omit<MotionProps, "animate"> {
 export const Alert = React.forwardRef<HTMLDivElement, AlertProps>(
   ({ variant, color, icon, open, action, onClose, animate, className, children, ...rest }, ref) => {
     // 1. init
-    const { alert } = useTheme();
-    const { defaultProps, valid, styles } = alert;
+    const { defaultProps, valid, styles } = useTheme("alert");
```
Comment (aakash-allen-forced) [inline packages/material-tailwind-react/src/components/Alert/index.tsx:58]: why this change across for usage of useTheme()?

**Review by aakash-allen-forced — APPROVED** [review]

File: packages/material-tailwind-react/src/components/Alert/index.tsx (hunk 2 above)
Comment (dinesh-kumar-allen) [reply packages/material-tailwind-react/src/components/Alert/index.tsx:58]: By default library was using all components' theme constant resulting in 20Kb of constant just loaded where we are using less than 1kB, so migrated the logic to inject the theme via theme-provider on-demand.
For new components, If the theme is not injected, we will throw an error so the developer can add the theme & start using it, which would ease the load on the root level.
